### ISSUE TITLE
Incorrect scene thumbnail when saving from script editor

### ISSUE THREAD

**Issue by nobuyukinyuu** [godotengine/godot#33808]
**Godot version:**
3.1.1

**OS/device including version:**
Windows 10 x64

**Issue description:**
When saving scenes from the script editor using Ctrl+S,  the thumbnail used to preview the scene in the file browser uses the last thing the editor viewport saw, even if it was from a different scene.  With multiple tabs, switching between code editors means overwriting the thumbnails of the associated scenes with the preview of the last scene which was viewed from the 2d editor before switching to the script editor.  (I haven't tested this with the 3d editor but assume it does the same.)

**Comment by lukruh** [comment]
I can confirm this on 3.1.1 stable on Linux. Generating thumbnails manually (#17932) could "solve" this bug as well.

**Comment by tom-jk** [comment]
Can reproduce in 3.1.1, 3.2, 3.2.2, 3.2.3.beta1 and 3.2.3.rc3.

Convenience project (rgb coloured sprite and sprite3d scenes, 3 each): [33808.zip](https://github.com/godotengine/godot/files/5028826/33808.zip)

Happens for both 2D and 3D scenes. 2D scene thumbnails will be overwritten with the last-seen image from 2D view mode, likewise 3D.

However, from 3.2.3.beta1, new 3D thumbnails aren't made at all.

**Comment by Deozaan** [comment]
This is still happening for me in 3.3.1-stable.

https://user-images.githubusercontent.com/227734/119252392-38923c00-bb69-11eb-8f32-f6bfaa3bd13e.mp4

I didn't understand the cause or the extent of the problem when I recorded this video, and discovered while recording it that it was worse than I thought. Then I came here to report the problem and found this issue which appears to match what I'm seeing.

**Comment by donn-xx** [comment]
Am on Godot 4.0.2 on Linux (Ubuntu) and am seeing these thumbnails/previews get worse over time. It also happens for the preview for @exported PackedScenes in the inspector. 

It seems to start right, but the longer Godot runs, the weirder they become.

Not sure how to demonstrate this. I would have to leave OBS recording and then video edit.

**Comment by Calinou** [comment]
See https://github.com/godotengine/godot-proposals/issues/4759, which will likely be needed to resolve this situation in a deterministic manner (or an approach inspired by it).

**Comment by donn-xx** [comment]
Calinou, I am finding that thumbnails from completely different scenes will appear. The main culprit is the preview of a @export var foo:PackedScene in the inspector. So the problem is more than a camera position in a scene.

**Comment by Deozaan** [comment]
I agree with donn-xx. As shown in my video above, thumbnails from entirely different scenes are saved. It appears as though whatever was the last displayed viewport scene is what gets saved as a thumbnail, even if you've changed scenes from within the script editor viewport. According to an earlier comment from tom-jk, 3D scenes get thumbnails from the last viewed 3D scene viewport, and 2D scenes get thumbnails from the last viewed 2D scene viewport.

While that linked proposal might be useful to get consistently nice thumbnails for each scene, even _bad thumbnails of the **correct** scene_ would be an improvement over the current behavior.
